### PR TITLE
Use rsIndexer instead of indexing manually

### PR DESCRIPTION
**Pull request by mborsz**
The goal is to simplify the code by reusing existing blocks: there already exists a rsIndexer that allows us to store objects along with some indexes for them (this is what is used by informers).

We can use here to avoid manual map operations.

/assign @tosi3k

### CODE REVIEW

File: clusterloader2/pkg/measurement/util/controlled_pods_indexer.go [hunk 1]
```diff
@@ -253,8 +257,16 @@ func (p *ControlledPodsIndexer) PodsControlledBy(obj interface{}) ([]*corev1.Pod
...
+		}
+		for _, replicaSet := range replicaSets {
+			replicaSet, ok := replicaSet.(*appsv1.ReplicaSet)
+			if !ok {
+				return nil, fmt.Errorf("expected *appsv1.ReplicaSet; got: %T", replicaSet)
+			}
+			podOwners = append(podOwners, replicaSet.GetUID())
 		}
 		p.UIDLock.Unlock()
 	default:
```
Comment (tosi3k) [inline clusterloader2/pkg/measurement/util/controlled_pods_indexer.go:272]: QQ: does the `default` path would work for bare `ReplicaSet` objects controlled by CL2?

If not, could we open an issue for that and link it here in a comment for documentation's sake?

**Review by tosi3k — CHANGES_REQUESTED** [review]

File: clusterloader2/pkg/measurement/util/controlled_pods_indexer.go (hunk 1 above)
Comment (mborsz) [reply clusterloader2/pkg/measurement/util/controlled_pods_indexer.go:272]: I think it should -- the pods set the owner directly so we should be able to get them the same way we get for deployment (deployments resolves to resplicasets, replicasets to pods)

**Pull request merged**